### PR TITLE
Feature/outputformats

### PR DESCRIPTION
**Pull request by KentMossback**
Serialize table response to choosen outputformat.  Read default outputformat from appsettings.

### CODE REVIEW

File: PxWeb/Code/Api2/Serialization/Csv2DataSerializer.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using Microsoft.AspNetCore.Http;
+using PCAxis.Paxiom;
+
+namespace PxWeb.Code.Api2.Serialization
+{
+    public class Csv2DataSerializer : IDataSerializer
+    {
+        public void Serialize(PXModel model, HttpResponse response)
+        {
+            response.ContentType = "text/csv; charset=" + System.Text.Encoding.Default.WebName;
```
Comment (MikaelNordberg) [inline PxWeb/Code/Api2/Serialization/Csv2DataSerializer.cs:10]: You should set charset according to how its done in the serializers. For example see link: https://github.com/statisticssweden/PCAxis.Core/blob/d1035b8dbe7f897544e4d5df40844f4fb96d3e45/PCAxis.Core/Serializers/CsvFileSerializer.vb#L42

File: PxWeb/Code/Api2/Serialization/XlsxDoubleColumnDataSerializer.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+using PCAxis.Paxiom;
+using PCAxis.Serializers;
+
+namespace PxWeb.Code.Api2.Serialization
+{
+    public class XlsxDoubleColumnDataSerializer : IDataSerializer
+    {
+        public void Serialize(PXModel model, HttpResponse response)
+        {
+            response.ContentType = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet; charset=" + System.Text.Encoding.Default.WebName;
```
Comment (MikaelNordberg) [inline PxWeb/Code/Api2/Serialization/XlsxDoubleColumnDataSerializer.cs:11]: Content type is not needed for the binary formats for example Excel. You only need to set charset.

Comment (KentMossback) [reply PxWeb/Code/Api2/Serialization/XlsxDoubleColumnDataSerializer.cs:11]: It won't work if that part is left out. Will leave it as it is.

Comment (likp) [reply PxWeb/Code/Api2/Serialization/XlsxDoubleColumnDataSerializer.cs:11]: It is only the encoding that can be omitted for the binary formats, you will still need the content-type.

**Review by likp — APPROVED** [review]

File: PxWeb/Code/Api2/Serialization/XlsxDataSerializer.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+using PCAxis.Paxiom;
+using PCAxis.Serializers;
+
+namespace PxWeb.Code.Api2.Serialization
+{
+    public class XlsxDataSerializer : IDataSerializer
+    {
+        public void Serialize(PXModel model, HttpResponse response)
+        {
+            response.ContentType = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet; charset=" + System.Text.Encoding.Default.WebName;
```
[the remark this answers is not in the document]
Comment (KentMossback) [reply PxWeb/Code/Api2/Serialization/XlsxDataSerializer.cs]: Removed charset as suggeted

**Review by MikaelNordberg — APPROVED** [review]
Looks good

**Review by likp — APPROVED** [review]

**Pull request merged**